scripts/lint.sh

```diff
@@ -7,50 +7,28 @@ if [[ $(flutter format -n .) ]]; then
 fi
 
 flutter pub get
-result=$(dartanalyzer lib/)
-if ! echo "$result" | grep -q "No issues found!"; then
-  echo "$result"
-  echo "dartanalyzer issue: lib"
-  exit 1
-fi
+
+# We need to run pubget on all the examples
+for f in doc/examples/**/pubspec.yaml; do
+  d=$(dirname $f)
+  cd $d
+  flutter pub get
+  cd - > /dev/null
+done
 
 analyzer() {
   cd $1
   flutter pub get
-  result=$(dartanalyzer .)
+  result=$(flutter analyze .)
   if ! echo "$result" | grep -q "No issues found!"; then
     echo "$result"
-    echo "dartanalyzer issue: $1"
+    echo "flutter analyze issue: $1"
     exit 1
   fi
   cd - > /dev/null
 }
 
-analyzer "example"
-
-#  Examples that are changed
-changed=$(git diff --name-only origin/develop doc/examples \
-  | xargs -I {} dirname {} | sed 's/\/lib$//' | uniq \
-  | xargs -I {} find {} -name pubspec.yaml | xargs -I {} dirname {})
-
-# Examples that are affected by changed code
-affected=$(git diff --name-only origin/develop lib/ \
-  | xargs -I {} basename {} | xargs -I {} grep -r -l --include \*.dart {} doc/examples/ \
-  | xargs -I {} dirname {} | sed 's/\/lib$//' | uniq \
-  | xargs -I {} find {} -name pubspec.yaml | xargs -I {} dirname {})
-
-both=("${changed[@]}" "${affected[@]}")
-lint_examples=$(printf "%s\n" "${both[@]}" | sort -u)
-for d in $lint_examples; do
-  analyzer $d
-done
-
-for f in doc/examples/**/pubspec.yaml; do
-  d=$(dirname $f)
-  if [[ ! " ${lint_examples[@]} " =~ " ${d} " ]]; then
-    analyzer $d
-  fi
-done
+analyzer .
```

Comment:
I think we can "inline" it now that there is only one callee.

I'm glad it worked out even with examples having custom deps!